app/controllers/api/message_channel_controller.rb

```diff
@@ -4,13 +4,13 @@ class API::MessageChannelController < API::RestfulController
   rescue_from(MessageChannelService::UnknownChannelError) { |e| respond_with_standard_error e, 400 }
 
   def subscribe
-    @subscriptions = [MessageChannelService.subscribe_to(user: current_user, channel: params[:channel])]
+    @subscriptions = [MessageChannelService.subscribe_to(user: current_user, model: subscription_model)] if subscription_model
     respond_with_subscriptions
   end
 
   def subscribe_user
-    @subscriptions = current_user.groups.map { |group| MessageChannelService.subscribe_to(user: current_user, channel: channel_for(group)) }
-    @subscriptions <<                                  MessageChannelService.subscribe_to(user: current_user, channel: "/notifications-#{current_user.id}")
+    @subscriptions = current_user.groups.map { |group| MessageChannelService.subscribe_to(user: current_user, model: group) }
```

Comment:
Yeah, works like a charm; it's nicely designed in that the frontend just listens to what the server tells it to, it has no concept of what the channels are named.